src/coreclr/src/tools/Common/Compiler/DependencyAnalysis/BlobNode.cs

```diff
@@ -34,9 +35,18 @@ public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
         public int Offset => 0;
         public override bool IsShareable => true;
 
+        public void SetRelocs(Relocation[] relocs)
```

Comment:
This is a rather dangerous API - it changes what the node depends on within the dependency graph. If the node is already marked, the new dependencies won't be visited. We try to avoid APIs like this.

I would prefer we create a new class for this (e.g. `SettableBlobNode`) that is pretty much a copy of `BlobNode` with these differences:

* Instead of the `byte[] _data` field, it has a `ObjectData _data` field.
* Data is not passed to the constructor (so we start with `_data == null`)
* `StaticDependenciesAreComputed` property is implemented as `return _data != null` (this is the protection to make sure we don't end up in a situation where first there are no dependencies reported in the graph)
* There is a `InitializeData(ObjectData data)` method to initialize the _data field.

Reply:
Thanks for your comment. I fixed based on your guide.